ui/src/app/views/breadcrumb/component.ts

```diff
@@ -0,0 +1,121 @@
+import {Component} from '@angular/core';
+import {ActivatedRoute, NavigationEnd, PRIMARY_OUTLET, Router} from '@angular/router';
+
+export interface Breadcrumb {
+  label: string;
+  url: string;
+}
+
+@Component({
+  styleUrls: ['../../styles/buttons.css',
+    '../../styles/cards.css',
+    '../../styles/headers.css',
+    '../../styles/inputs.css',
+    './component.css'],
+  templateUrl: './component.html',
+  selector: 'app-breadcrumb'
+})
+
+export class BreadcrumbComponent {
+
+  breadcrumbs: Breadcrumb[];
+
+  constructor(
+      private activatedRoute: ActivatedRoute,
+      private router: Router) {
+    this.router.events.filter(event => event instanceof NavigationEnd).subscribe(event => {
```

Comment:
This should happen in an `ngOnInit` method, and should be assigned to a subscription property on the class that is unsubscribed from in an `ngOnDestroy` method.

Assuming we only ever want one breadcrumb component, and the lifecycle of that component should match the lifetime of the app as a whole, this won't lead to a memory leak, but it still seems better to explicitly manage the memory than to bank on those assumptions.

Reply:
We actually tried it in ngoninit, and that didn’t work, because it needed to subscribe before the router event fired, and flow was:
constructor -> routing -> ngoninit.
I agree with the destructor though.

Reply:
Let me see if I can make this work with the `Subscription` model. I'd like to follow consistent patterns, but it was beyond my angular-fu to make it work on page refresh.